Feature/#45 article wish

IMWoo94 opened the pull request:
이슈 : #45 [게시글] > 관심 목록 등록 기능

1. 게시글 관심 등록, 수정, 삭제, 조회 기능 구현 -> JPA
2. 게시글 관심 등록 Http 테스터 작성

IMWoo94 committed:
feature/#45-article-wish

이슈 : #45 [게시글] > 관심 목록 등록 기능

1. 게시글 관심 등록 처리 하기 위한 기능 추가 구현
2. 게시글 관심 등록 Http 테스터 작성
3. 불필요 service 선언 제거

IMWoo94 committed:
feature/#45-article-wish

이슈 : #45 [게시글] > 관심 목록 등록 기능

1. Objects.isNull -> null 확인으로 변경

IMWoo94 committed:
feature/#45-article-wish

이슈 : #45 [게시글] > 관심 목록 등록 기능

1. 관심목록 삭제 시 관심 목록 ID 만 넘어오면 삭제처리 진행이 되도록 변경

IMWoo94 committed:
feature/#45-article-wish

이슈 : #45 [게시글] > 관심 목록 등록 기능

1. 관심목록 기등록 건에 대해서 변경 이벤트 추가

IMWoo94 committed:
feature#54-code-cverage
이슈 : #54 Jacoco code coverage apply

1. jacoco 코드 커버리지 플러그인 등록
2. jacoco report(xml, html) 생성
3. jacoco 커버리지 기준 등록 ( 전체 10%, 클래스[ 구문 분석 1%, 분기 분석 1% ]

IMWoo94 committed:
feature#54-code-coverage
이슈 : #54 커밋 메시지 컨벤션 적용

커밋 컨벤션 적용

IMWoo94 committed:
feature#54-code-coverage
이슈 : #54 PR 시 코드 커버리지 코멘트 제공

PR 시 코드 커버리지 코멘트로 구성원간의 공유

IMWoo94 committed:
feature#54-code-coverage
이슈 : #54 코드 커버리지 Jacoco 적용

코드 커버리지 테스트 대상 중 MapStruct 클래스 파일 제외

IMWoo94 committed:
feature#54-code-coverage
이슈 : #54 코드 커버리지 Jacoco 적용

Github Actions Self-hosted runner 사용

IMWoo94 committed:
feature#54-code-coverage
이슈 : #54 코드 커버리지 Jacoco 적용

오타 수정

IMWoo94 committed:
feature#55-actions-cd
이슈 : #55 Github Actions CD 구현

Github Actions CD 구현 및
배포 스크립트 수정

f-lab-ben commented on src/main/java/com/danggeun/wish/service/WishService.java:
```diff
@@ -0,0 +1,48 @@
+	private final WishRespository wishRespository;
+
+	public WishResponseDto createWish(WishRequestDto wishRequestDto) {
+		WishResponseDto result = null;
```
null로 변수를 초기화 하고 메서드 단위에서 공용으로 사용하는걸 지양해주세요.
이렇게 되면 어느시점에 어떤값이 들어가는지 mutable해져서 위험해집니다.

차라리 if else에서 바로 찾아진 result를 return하는 구조로 짜는게 좋아요

IMWoo94 replied:
이 부분은 제가 놓친거 같아요.
왜 저기를 null 초기화를 했는지...
불변을 지킬 수 있도록 결과 바로 return 처리 하겠습니다.

IMWoo94 committed:
feature#55-actions-cd
이슈 : #55 Github Actions CD 구현

프로젝트 실행 시 profile = dev 로 실행 설정 추가

IMWoo94 commented on src/main/java/com/danggeun/wish/controller/WishController.java:
```diff
@@ -0,0 +1,61 @@
+	}
+
+	// 관심목록 수정
+	@PutMapping
```
별도의 DTO 에 대한 전체를 보내는 과정은 구현을 하면서 저도 약간 불편하다는 생각이 있었습니다.
ID 로 조회가 가능하다면 ID 하나만으로 해결하는 게 간편하다고 생각이 듭니다.

IMWoo94 committed:
feature#54-code-coverage
feat : #54 코드 커버리지 Jacoco 적용

Github actions Runner self-hosted -> github-hoste 변경
테스트 커버리지 등록 PR 의 경우에만 동작하도록 조건문 추가 및 Build 완료 시 동작 처리

IMWoo94 committed:
Merge branch 'feature#54-code-coverage' into feature#55-actions-cd

IMWoo94 committed:
feature#55-actions-cd
feat : #55 Github Actions CD 구현

build 완료 시 서버 경로에 Jar 파일 SCP 사용하여 전달
Jar 파일 구동 및 기존 서버 종료 스크립트 실행

IMWoo94 committed:
feature#55-actions-cd
feat : #55 Github Actions CD 구현

전체 Workflow 에 대해서 브랜치 별로 하나의 Runner 만 구동이 가능하도록 동시성 적용
배포 작업의 경우 하나의 Workflow 안에서 한번만 발생하도록 동시성 제어

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. 관심목록 업데이트 PathVariable 로 ID 만 전달

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. http test 작성

IMWoo94 commented:
이슈사항 해결 후 PR 재 오픈

IMWoo94 committed:
feature#54-code-coverage
feat : #54 코드 커버리지 Jacoco 적용

build application yml 파일 github actions secrets 로 생성 후 처리

IMWoo94 committed:
feature#54-code-coverage
feat : #54 코드 커버리지 Jacoco 적용

build application yml 파일 github actions secrets 로 생성 후 처리

f-lab-ben commented:
이 이슈는 진행상황이 어떻게 되나요?

IMWoo94 commented:
JPA 로 변경 후 재 오픈 및 반영 하도록 하겠습니다.

IMWoo94 committed:
Merge pull request #56 from f-lab-edu/feature#54-code-coverage

Feature#54 code coverage

IMWoo94 committed:
Merge pull request #59 from f-lab-edu/feature#55-actions-cd

Feature#55 actions cd

IMWoo94 committed:
Merge branch 'develop' into feature/#45-article-wish

IMWoo94 committed:
Merge pull request #63 from f-lab-edu/feature/#13-article-write

Feature/#13-article-write

IMWoo94 committed:
Merge branch 'develop' into feature/#45-article-wish

IMWoo94 committed:
feature/#45-article-wish

fix : #45 [게시글] > 관심 목록 등록 기능

1. 오타 수정

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. 게시글 관심 등록 기능 Jpa 로 변경

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. 게시글 관심 수정,삭제 기능 Jpa 로 변경

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. 게시글과 관심목록 연관관계 수정 다대일 관계로 변경

IMWoo94 committed:
feature/#45-article-wish

refactor : #45 [게시글] > 관심 목록 등록 기능

1. 관심 목록 조회 JPA 변경 및 오타 수정

IMWoo94 committed:
feature/#45-article-wish

test : #45 [게시글] > 관심 목록 등록 기능

1. 관심 목록 테스트 작성
